### PR TITLE
[CI] test signed binaries and released binaries

### PR DESCRIPTION
**Pull request by v1v**
### What

Since internal-ci is in charge to generate the signed artifacts and publish them to github, this PR verifies any binaries manipulation get tested `before` and `after` the release.

Reports the release status if test failures.

Verify signed binaries (deb, rpm and tar) with their asc files as stated in [here](https://serverfault.com/a/896232).

### Why

Detect any corruptions in the binaries and validates the installation process similar to what our users will run.

### Issues

Closes https://github.com/elastic/apm-agent-php/issues/191 and https://github.com/elastic/apm-agent-php/issues/190

### How to test this

```bash
$ RELEASE_VERSION=0.1 make -C packaging install-release-github
...
OK (19 tests, 600 assertions)
> @putenv ELASTIC_APM_TESTS_APP_CODE_HOST_KIND=CLI_builtin_HTTP_server
> composer run-script run_component_tests_configured
> phpunit -c phpunit_component_tests.xml
PHPUnit 8.5.8 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.33
Configuration: /src/phpunit_component_tests.xml

...................                                               19 / 19 (100%)

Time: 41.58 seconds, Memory: 14.00 MB

$ PHP_VERSIO=7.4 RELEASE_VERSION=0.1 make -C packaging install-release
```

### DEB

```
+ BUILD_RELEASES_FOLDER=build/releases
+ '[' release-github == deb ']'
+ '[' release-github == release-github ']'
+ PACKAGE=apm-agent-php_0.1-preview_all.deb
+ downloadWithShasumValidation apm-agent-php_0.1-preview_all.deb build/releases https://github.com/elastic/apm-agent-php/releases/download/v0.1
+ package=apm-agent-php_0.1-preview_all.deb
+ folder=build/releases
+ url=https://github.com/elastic/apm-agent-php/releases/download/v0.1
+ mkdir -p build/releases
+ wget -q https://artifacts.elastic.co/GPG-KEY-elasticsearch -O build/releases/GPG-KEY-elasticsearch
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php_0.1-preview_all.deb -O build/releases/apm-agent-php_0.1-preview_all.deb
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php_0.1-preview_all.deb.sha512 -O build/releases/apm-agent-php_0.1-preview_all.deb.sha512
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php_0.1-preview_all.deb.asc -O build/releases/apm-agent-php_0.1-preview_all.deb.asc
+ cd build/releases
+ gpg --import GPG-KEY-elasticsearch
gpg: directory '/root/.gnupg' created
gpg: keybox '/root/.gnupg/pubring.kbx' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key D27D666CD88E42B4: public key "Elasticsearch (Elasticsearch Signing Key) <dev_ops@elasticsearch.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
+ shasum -a 512 -c apm-agent-php_0.1-preview_all.deb.sha512
apm-agent-php_0.1-preview_all.deb: OK
+ gpg --verify apm-agent-php_0.1-preview_all.deb.asc apm-agent-php_0.1-preview_all.deb
gpg: Signature made Tue Sep 29 16:34:10 2020 UTC
gpg:                using RSA key 46095ACC8548582C1A2699A9D27D666CD88E42B4
gpg: Good signature from "Elasticsearch (Elasticsearch Signing Key) <dev_ops@elasticsearch.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 4609 5ACC 8548 582C 1A26  99A9 D27D 666C D88E 42B4
+ cd -
/src
+ dpkg-sig --verify build/releases/apm-agent-php_0.1-preview_all.deb
Processing build/releases/apm-agent-php_0.1-preview_all.deb...
GOODSIG _gpgelastic 46095ACC8548582C1A2699A9D27D666CD88E42B4 1601397249
+ dpkg -i build/releases/apm-agent-php_0.1-preview_all.deb
```

### RPM

```
+ BUILD_RELEASES_FOLDER=build/releases
+ '[' release-github == rpm ']'
+ '[' release-github == release-github ']'
+ PACKAGE=apm-agent-php-0.1_preview-1.noarch.rpm
+ download apm-agent-php-0.1_preview-1.noarch.rpm build/releases https://github.com/elastic/apm-agent-php/releases/download/v0.1
+ package=apm-agent-php-0.1_preview-1.noarch.rpm
+ folder=build/releases
+ url=https://github.com/elastic/apm-agent-php/releases/download/v0.1
+ mkdir -p build/releases
+ wget -q https://artifacts.elastic.co/GPG-KEY-elasticsearch -O build/releases/GPG-KEY-elasticsearch
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php-0.1_preview-1.noarch.rpm -O build/releases/apm-agent-php-0.1_preview-1.noarch.rpm
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php-0.1_preview-1.noarch.rpm.sha512 -O build/releases/apm-agent-php-0.1_preview-1.noarch.rpm.sha512
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php-0.1_preview-1.noarch.rpm.asc -O build/releases/apm-agent-php-0.1_preview-1.noarch.rpm.asc
+ cd build/releases
+ gpg --import GPG-KEY-elasticsearch
gpg: directory `/root/.gnupg' created
gpg: new configuration file `/root/.gnupg/gpg.conf' created
gpg: WARNING: options in `/root/.gnupg/gpg.conf' are not yet active during this run
gpg: keyring `/root/.gnupg/secring.gpg' created
gpg: keyring `/root/.gnupg/pubring.gpg' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key D88E42B4: public key "Elasticsearch (Elasticsearch Signing Key) <dev_ops@elasticsearch.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
+ shasum -a 512 -c apm-agent-php-0.1_preview-1.noarch.rpm.sha512
apm-agent-php-0.1_preview-1.noarch.rpm: OK
+ gpg --verify apm-agent-php-0.1_preview-1.noarch.rpm.asc apm-agent-php-0.1_preview-1.noarch.rpm
gpg: Signature made Tue Sep 29 16:34:08 2020 UTC using RSA key ID D88E42B4
gpg: Good signature from "Elasticsearch (Elasticsearch Signing Key) <dev_ops@elasticsearch.org>"
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 4609 5ACC 8548 582C 1A26  99A9 D27D 666C D88E 42B4
+ cd -
/src
+ rpm -ivh build/releases/apm-agent-php-0.1_preview-1.noarch.rpm
warning: build/releases/apm-agent-php-0.1_preview-1.noarch.rpm: Header V4 RSA/SHA512 Signature, key ID d88e42b4: NOKEY
```

### TAR

```
+ BUILD_RELEASES_FOLDER=build/releases
+ '[' release-tar-github == rpm ']'
+ '[' release-tar-github == release-github ']'
+ '[' release-tar-github == release-tar-github ']'
+ PACKAGE=apm-agent-php.tar
+ download apm-agent-php.tar build/releases https://github.com/elastic/apm-agent-php/releases/download/v0.1
+ package=apm-agent-php.tar
+ folder=build/releases
+ url=https://github.com/elastic/apm-agent-php/releases/download/v0.1
+ mkdir -p build/releases
+ wget -q https://artifacts.elastic.co/GPG-KEY-elasticsearch -O build/releases/GPG-KEY-elasticsearch
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php.tar -O build/releases/apm-agent-php.tar
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php.tar.sha512 -O build/releases/apm-agent-php.tar.sha512
+ wget -q https://github.com/elastic/apm-agent-php/releases/download/v0.1/apm-agent-php.tar.asc -O build/releases/apm-agent-php.tar.asc
+ cd build/releases
+ gpg --import GPG-KEY-elasticsearch
gpg: directory `/root/.gnupg' created
gpg: new configuration file `/root/.gnupg/gpg.conf' created
gpg: WARNING: options in `/root/.gnupg/gpg.conf' are not yet active during this run
gpg: keyring `/root/.gnupg/secring.gpg' created
gpg: keyring `/root/.gnupg/pubring.gpg' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key D88E42B4: public key "Elasticsearch (Elasticsearch Signing Key) <dev_ops@elasticsearch.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
+ shasum -a 512 -c apm-agent-php.tar.sha512
apm-agent-php.tar: OK
+ gpg --verify apm-agent-php.tar.asc apm-agent-php.tar
gpg: Signature made Tue Sep 29 16:34:08 2020 UTC using RSA key ID D88E42B4
gpg: Good signature from "Elasticsearch (Elasticsearch Signing Key) <dev_ops@elasticsearch.org>"
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 4609 5ACC 8548 582C 1A26  99A9 D27D 666C D88E 42B4
+ cd -
/src
+ tar -xf build/releases/apm-agent-php.tar -C /
+ source /.scripts/after_install


```

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -307,6 +308,41 @@ pipeline {
...
+            }
+            stages {
+              stage('Release Test') {
+                steps {
+                  withGithubNotify(context: "Signed-Test-${PHP_VERSION}") {
+                    deleteDir()
+                    unstash 'source'
+                    dir("${BASE_DIR}") {
+                      unstash "${env.SIGNED_ARTIFACTS}"
+                      sh script: "PHP_VERSION=${PHP_VERSION} -C packaging install", label: 'package install'
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:330]:
```suggestion
                      sh script: "PHP_VERSION=${PHP_VERSION} make -C packaging install", label: 'package install'
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -325,19 +361,53 @@ pipeline {
...
+              }
+            }
+            stages {
+              stage('Release Test') {
+                steps {
+                  withGithubNotify(context: "Release-Test-${PHP_VERSION}") {
+                    deleteDir()
+                    unstash 'source'
+                    dir("${BASE_DIR}"){
+                      sh script: "PHP_VERSION=${PHP_VERSION} RELEASE_VERSION=${env.TAG_NAME.replaceAll('^v', '')} -C packaging install-release-github", label: 'package install-release-github'
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:386]:
```suggestion
                      sh script: "PHP_VERSION=${PHP_VERSION} RELEASE_VERSION=${env.TAG_NAME.replaceAll('^v', '')} make -C packaging install-release-github", label: 'package install-release-github'
```

**Review by mdelapenya — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -268,6 +268,7 @@ pipeline {
         SIGNED_ARTIFACTS = 'signed-artifacts'
         BUCKET_SUBFOLDER_SIGNED_ARTIFACTS = "${env.BUCKET_SUBFOLDER}/${env.SIGNED_ARTIFACTS}"
         BUCKET_SIGNED_ARTIFACTS_PATH = "gs://${env.BUCKET_NAME}/${env.BUCKET_SUBFOLDER_SIGNED_ARTIFACTS}"
+        RELEASE_URL_MESSAGE = "(<https://github.com/elastic/apm-agent-php/releases/tag/${env.TAG_NAME}|${env.TAG_NAME}>)"
```
Comment (SergeyKleyman) [inline .ci/Jenkinsfile:271]: `${env.TAG_NAME}` is repeated - is it intentional?

**Review by SergeyKleyman — APPROVED** [review]

File: .ci/Jenkinsfile (hunk 3 above)
Comment (v1v) [reply .ci/Jenkinsfile:271]: Yes.

Notifications are happening from two different channels (slack and email) so, the current format is based on the slack format so links are crated with the format `(<url|description>)`. Email body is transformed based on that particular pattern.

**Pull request merged**